includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -1522,4 +1569,446 @@ public function add_twentyfourteen_search() {
+			return;
+		}
+
+		$body_id = $this->get_element_id( $this->get_body_node(), 'body' );
+
+		$open_xpaths  = isset( $args['open_button_xpath'] ) ? $args['open_button_xpath'] : [];
+		$close_xpaths = isset( $args['close_button_xpath'] ) ? $args['close_button_xpath'] : [];
+
+		$buttons = [
+			"{$modal_id}.open"  => $open_xpaths,
+			"{$body_id}.toggleClass(class=showing-modal,force=true)" => $open_xpaths,
+			"{$modal_id}.close" => $close_xpaths,
+			"{$body_id}.toggleClass(class=showing-modal,force=false)" => $close_xpaths,
+		];
+
+		foreach ( $buttons as $action => $button_xpaths ) {
+			foreach ( $button_xpaths as $button_xpath ) {
+				foreach ( $this->xpath->query( $button_xpath ) as $node ) {
+					if ( $node instanceof DOMElement ) {
+						$this->add_amp_action( $node, 'tap', $action );
+					}
+				}
+			}
+		}
+
+		// Create an <amp-lightbox> element that will contain the modal.
+
+		$amp_lightbox = $this->dom->createElement( 'amp-lightbox' );
+		$amp_lightbox->setAttribute( 'id', $modal_id );
+		$amp_lightbox->setAttribute( 'layout', 'nodisplay' );
+		$amp_lightbox->setAttribute( 'animate-in', isset( $args['animate_in'] ) ? $args['animate_in'] : 'fade-in' );
+		$amp_lightbox->setAttribute( 'scrollable', isset( $args['scrollable'] ) ? $args['scrollable'] : true );
+
+		$parent_node = $modal_content_node->parentNode;
+		$parent_node->replaceChild( $amp_lightbox, $modal_content_node );
+
+		$strip_wrapper_levels = isset( $args['strip_wrapper_levels'] ) ? $args['strip_wrapper_levels'] : 0;
+
+		while ( $strip_wrapper_levels > 0 ) {
+			$children = [];
+			foreach ( $modal_content_node->childNodes as $child_node ) {
+				if ( $child_node instanceof DOMElement && ! $child_node instanceof DOMComment ) {
+					$children[] = $child_node;
+				}
+			}
+
+			if ( count( $children ) > 1 ) {
+				break;
+			}
+
+			// Add class(es) of removed wrapper to lightbox to avoid breaking CSS selectors.
+			if ( $modal_content_node->hasAttribute( 'class' ) ) {
+				$classes = $modal_content_node->getAttribute( 'class' );
+				if ( $amp_lightbox->hasAttribute( 'class' ) ) {
+					$classes .= ' ' . $amp_lightbox->getAttribute( 'class' );
+				}
+				$amp_lightbox->setAttribute( 'class', $classes );
+			}
+
+			$modal_content_node = $modal_content_node->removeChild( $children[0] );
+
+			$strip_wrapper_levels--;
+		}
+
+		$amp_lightbox->appendChild( $modal_content_node );
+	}
+
+	/**
+	 * Add generic modal interactivity compat for the Twentytwenty theme.
+	 *
+	 * Modals implemented in JS will be transformed into <amp-lightbox> equivalents,
+	 * with the tap actions being attached to their associated toggles.
+	 */
+	public function add_twentytwenty_modals() {
+		$modals = $this->xpath->query( "//*[ @class and contains( concat( ' ', normalize-space( @class ), ' ' ), ' cover-modal ' ) ]" );
+
+		if ( false === $modals || 0 === $modals->count() ) {
+			return;
+		}
+
+		foreach ( $modals as $modal ) {
+
+			if ( ! $modal instanceof DOMElement ) {
```

Comment:
The syntax you've linked to is for the `element()` node-type test, i.e. "from a given set of nodes, only return the ones of type element".

The asterisk I'm using in my XPath is actually this one in the syntax: https://www.w3.org/TR/xpath-31/#doc-xpath31-Wildcard
It is then followed by a predicate: https://www.w3.org/TR/xpath-31/#doc-xpath31-Predicate to denote which nodes to retrieve.

I'm only getting nodes of type element because the full query (in unabreviated syntax) would be:
```
child:://*[ attribute::class and contains( concat( ' ', normalize-space( attribute::class ), ' ' ), ' cover-modal ' ) ]
```
So, generally, if no axis provided, the `child` axis is the default... most of the time:
```
If the axis name is omitted from an axis step, the default axis is child, with
two exceptions: (1) if the NodeTest in an axis step contains an AttributeTest or
SchemaAttributeTest then the default axis is attribute; (2) if the NodeTest in
an axis step is a NamespaceNodeTest then the default axis is namespace - in an
implementation that does not support the namespace axis, an error is raised
[err:XQST0134].
```
This means, in this specific case, I'm operating on the `child` axis, and the principal node type of that axis is the `element` (see https://www.w3.org/TR/xpath-31/#dt-axis-step).

This means I do indeed get only element nodes, but it is more complex than just having the `*` wildcard.

Also, the alternative (for proper IDE support) is to have this instead:
![Image 2019-10-15 at 9 29 02 AM](https://user-images.githubusercontent.com/83631/66809734-3ea44d80-ef2e-11e9-819a-8aeb544c8600.png)

I'm changing the checks into this docblock then (which is forced to be unnecessarily cumbersome by PHPCS), as you can argue that the actual check incurs a performance overhead. However, I'm not sure that is a big deal, and given the complexity of the issue, and the obscene alternative, this could just be considered defensive coding.

Reply:
OK, I relent 😄 Carry on.

Reply:
The checks were already replaced by docblock atrocities in https://github.com/ampproject/amp-wp/pull/3403/commits/6ce943b3875c3348963be2a8c056e298dbf278a3

I hope you're happy with yourself! 😉